visualization/tools/pcd_viewer.cpp

```diff
@@ -678,17 +685,281 @@ main (int argc, char** argv)
       print_info ("Camera parameters restored from %s.\n", p->getCameraFile ().c_str ());
   }
 
+  pcl::PLYReader ply;
+    // Go through PCD files
+  for (size_t i = 0; i < ply_file_indices.size (); ++i)
+  {
+    tt.tic ();
+    cloud.reset (new pcl::PCLPointCloud2);
+    Eigen::Vector4f origin;
+    Eigen::Quaternionf orientation;
+    int version;
+
+    print_highlight (stderr, "Loading "); print_value (stderr, "%s ", argv[ply_file_indices.at (i)]);
+
+    if (ply.read (argv[ply_file_indices.at (i)], *cloud, origin, orientation, version) < 0)
+      return (-1);
+
+      // Calculate transform if available.
+    if (pose_x.size () > i && pose_y.size () > i && pose_z.size () > i &&
+        pose_roll.size () > i && pose_pitch.size () > i && pose_yaw.size () > i)
+    {
+      Eigen::Affine3f pose =
+        Eigen::Translation3f (Eigen::Vector3f (pose_x[i], pose_y[i], pose_z[i])) *
+        Eigen::AngleAxisf (pose_yaw[i],   Eigen::Vector3f::UnitZ ()) *
+        Eigen::AngleAxisf (pose_pitch[i], Eigen::Vector3f::UnitY ()) *
+        Eigen::AngleAxisf (pose_roll[i],  Eigen::Vector3f::UnitX ());
+      orientation = pose.rotation () * orientation;
+      origin.block<3, 1> (0, 0) = (pose * Eigen::Translation3f (origin.block<3, 1> (0, 0))).translation ();
+    }
+
+    std::stringstream cloud_name;
+
+    // ---[ Special check for 1-point multi-dimension histograms
```

Comment:
I don't think anyone stores 1-point multi-dimensional histograms (whatever that is :laughing:) in PLY files. This is only relevant for PCDs.

Reply:
As you say, whatever that is, this block should be dedicated to PCD and discard for PLY files?

Reply:
Yep. The common use of ply is for meshed objects. If someone is storing histograms it's probably because it came from a PCL environment already and for those cases PCD is the way to go.

Reply:
Oh, I didn't know that PLY file are more appropriate for meshed objects. Do you have a link about that please?

Regarding the code, I think that we should have a function which deals with any specificities of PCD files. I'll do a proposition soon.

Reply:
What I wrote was that **the common use of ply** is for meshed objects. Google for "ply file repository" and see the types of search results you get.